samples/examples/Reggression_BikeSharingDemands/BikeSharingDemand/Program.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Linq;
+using System.Runtime.InteropServices;
+using BikeSharingDemand.BikeSharingDemandData;
+using BikeSharingDemand.Helpers;
+using BikeSharingDemand.Model;
+using Microsoft.ML;
+using Microsoft.ML.Models;
+using Microsoft.ML.Trainers;
+
+namespace BikeSharingDemand
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            var trainingDataLocation = @"Data/hour_train.csv";
+            var testDataLocation = @"Data/hour_test.csv";
+
+            var modelEvaluator = new ModelEvaluator();
+
+            var fastTreeModel = new ModelBuilder(trainingDataLocation, new FastTreeRegressor()).BuildAndTrain();
+            var fastTreeMetrics = modelEvaluator.Evaluate(fastTreeModel, testDataLocation);
+            PrintMetrics("Fast Tree", fastTreeMetrics);
+
+            var fastForestModel = new ModelBuilder(trainingDataLocation, new FastForestRegressor()).BuildAndTrain();
+            var fastForestMetrics = modelEvaluator.Evaluate(fastForestModel, testDataLocation);
+            PrintMetrics("Fast Forest", fastForestMetrics);
+
+            var poissonModel = new ModelBuilder(trainingDataLocation, new PoissonRegressor()).BuildAndTrain();
+            var poissonMetrics = modelEvaluator.Evaluate(poissonModel, testDataLocation);
+            PrintMetrics("Poisson", poissonMetrics);
+
+            var gradientDescentModel = new ModelBuilder(trainingDataLocation, new OnlineGradientDescentRegressor()).BuildAndTrain();
+            var gradientDescentMetrics = modelEvaluator.Evaluate(gradientDescentModel, testDataLocation);
+            PrintMetrics("Online Gradient Descent", gradientDescentMetrics);
+
+            var fastTreeTweedieModel = new ModelBuilder(trainingDataLocation, new FastTreeTweedieRegressor()).BuildAndTrain();
+            var fastTreeTweedieMetrics = modelEvaluator.Evaluate(fastTreeTweedieModel, testDataLocation);
+            PrintMetrics("Fast Tree Tweedie", fastTreeTweedieMetrics);
+
+            var additiveModel = new ModelBuilder(trainingDataLocation, new GeneralizedAdditiveModelRegressor()).BuildAndTrain();
+            var additiveMetrics = modelEvaluator.Evaluate(additiveModel, testDataLocation);
+            PrintMetrics("Generalized Additive Model", additiveMetrics);
+
+            var stohasticDualCorordinateAscentModel = new ModelBuilder(trainingDataLocation, new StochasticDualCoordinateAscentRegressor()).BuildAndTrain();
+            var stohasticDualCorordinateAscentMetrics = modelEvaluator.Evaluate(stohasticDualCorordinateAscentModel, testDataLocation);
+            PrintMetrics("Stochastic Dual Coordinate Ascent", stohasticDualCorordinateAscentMetrics);
+
+            VisualizeTenPredictionsForTheModel(fastTreeTweedieModel, testDataLocation);
+            fastTreeTweedieModel.WriteAsync(@".\Model.zip");

```

Comment:
I guess you are generating the .ZIP model file for the FastTreeTweedie learner because it has the best "R2 Score == 0.92". 
As a "nice to have", it would be nice to either generate one .ZIP file per approach/learner with a different name (just like the "ML.NET Model Builder UI app" does when doing multiple trainings) or if you generate a single model .ZIP file, it would be good to instead of doing it "hard-coded", you'd compare the metrics and decide to generate the model based on that comparison and generating the best one, dynamically, not hard-coded for the FastTreeTweedie learner. - This is just an idea/improvement, not really needed. 👍